improve notification app integration, use upstream notifications app instead of outdated fork (separate branch)

cezio opened the pull request:
This updates notifications integration code:

moves integration logic to single point: geonode.notifications_helper
use AppConfig.ready() based integration,
update template paths

(replaces #3045)

simod commented:
Hi this looks very interesting. Just a note, why the changes in the CSS and js files? Also Travis is not very happy.

cezio commented:
@simod i'm not sure actually, i was updating statics with grunt in the meantime, must have slipped those changes in somehow.

I'm trying to fix travis. that's a bit more complicated, because this code depends on changes i made to django-user-account, which are not merged yet. So i've added temporary fix: separate branch with fixes (https://github.com/cezio/django-user-accounts/tree/252_255_mixed) and updated deps in geonode

afabiani commented on geonode/notifications_helper.py:
```diff
@@ -0,0 +1,109 @@
+#########################################################################
+#
+# Copyright (C) 2016 OSGeo
+#
```
New files should have Copyright to 2017

afabiani commented on geonode/settings.py:
```diff
@@ -330,7 +330,7 @@
 
     # Theme
-    "pinax_theme_bootstrap_account",
+    #"pinax_theme_bootstrap_account",
```
Is it correct to remove this dependency here?

cezio replied:
fixed

afabiani commented on geonode/settings.py:
```diff
@@ -1029,6 +1029,12 @@
+# or notification
+NOTIFICATIONS_MODULE = 'pinax.notifications'
+
+if TEST:
```
Is this correct? What TEST is?

afabiani commented on geonode/static/lib/css/L.Control.Pan.css:
```diff
@@ -85,14 +85,14 @@
 
 .leaflet-control-pan-up {
-	background-image: url(../img/pan-up.png);
+	background-image: url('../img/../img/pan-up.png');
```
This should not be modified

cezio replied:
fixed (along with other static files)

afabiani commented on geonode/static/lib/css/L.Control.Pan.css:
```diff
@@ -85,14 +85,14 @@
 }
 .leaflet-control-pan-left {
-	background-image: url(../img/pan-left.png);
+	background-image: url('../img/../img/pan-left.png');
```
This should not be modified

afabiani commented on geonode/static/lib/css/L.Control.Pan.css:
```diff
@@ -85,14 +85,14 @@
 }
 .leaflet-control-pan-right {
-	background-image: url(../img/pan-right.png);
+	background-image: url('../img/../img/pan-right.png');
```
This should not be modified

afabiani commented on geonode/static/lib/css/L.Control.Pan.css:
```diff
@@ -85,14 +85,14 @@
 }
 .leaflet-control-pan-down {
-	background-image: url(../img/pan-down.png);
+	background-image: url('../img/../img/pan-down.png');
```
This should not be modified

afabiani commented on geonode/static/lib/css/jquery.tree-multiselect.min.css:
```diff
@@ -1,2 +1,2 @@
-/* jQuery Tree Multiselect v2.1.3 | (c) Patrick Tsai | MIT Licensed */
-div.tree-multiselect{border:2px solid #D8D8D8;border-radius:5px;display:table;height:inherit;width:100%}div.tree-multiselect>div.selected,div.tree-multiselect>div.selections{display:inline-block;vertical-align:top;box-sizing:border-box;-moz-box-sizing:border-box;-webkit-box-sizing:border-box;display:table-cell;padding:1%;width:50%}div.tree-multiselect>div.selections{border-right:solid 2px #D8D8D8 ...
+/* jQuery Tree Multiselect v2.2.1 | (c) Patrick Tsai | MIT Licensed */
```
This file should not be modified

afabiani commented on geonode/static/lib/css/leaflet.fullscreen.css:
```diff
@@ -1,5 +1,5 @@
 .leaflet-control-fullscreen a {
-  background:#fff url(fullscreen.png) no-repeat 0 0;
+  background:#fff url('../img/fullscreen.png') no-repeat 0 0;
```
This looks a bit strange, can you double check?

afabiani commented on geonode/static/lib/css/leaflet.fullscreen.css:
```diff
@@ -35,6 +35,6 @@
   (min-resolution:192dpi) {
     .leaflet-control-fullscreen a {
-      background-image:url(fullscreen@2x.png);
+      background-image:url('../img/fullscreen@2x.png');
```
This looks a bit strange, can you double check?

afabiani commented on geonode/static/lib/js/fastselect.standalone.min.js:
```diff
@@ -1,2 +1,2 @@
-!function(a){"function"==typeof define&&define.amd?define(["jquery"],a):"object"==typeof module&&module.exports?module.exports=a(require("jquery")):a(jQuery)}(function(a){function b(a,b){this.init.apply(this,arguments)}var c=a(window.document),d=0,e={13:"enter",27:"escape",40:"downArrow",38:"upArrow"};return a.extend(b.prototype,{init:function(c,e){e=this.options=a.extend(!0,{},b.defaults,e),this ...
+!function(a){"function"==typeof define&&define.amd?define(["jquery"],a):"object"==typeof module&&module.exports?module.exports=a(require("jquery")):a(jQuery)}(function(a){function b(a,b){this.init.apply(this,arguments)}var c=a(window.document),d=0,e=/\w\b/g,f={13:"enter",27:"escape",40:"downArrow",38:"upArrow"};return a.extend(b.prototype,{init:function(c,e){e=this.options=a.extend(!0,{},b.defaul ...
```
Why this file has changed?

afabiani commented on geonode/static/lib/js/jquery.tree-multiselect.min.js:
```diff
@@ -1,2 +1,2 @@
-/* jQuery Tree Multiselect v2.1.3 | (c) Patrick Tsai | MIT Licensed */
-!function a(b,c,d){function e(g,h){if(!c[g]){if(!b[g]){var i="function"==typeof require&&require;if(!h&&i)return i(g,!0);if(f)return f(g,!0);var j=new Error("Cannot find module '"+g+"'");throw j.code="MODULE_NOT_FOUND",j}var k=c[g]={exports:{}};b[g][0].call(k.exports,function(a){var c=b[g][1][a];return e(c?c:a)},k,k.exports,a,b,c,d)}return c[g].exports}for(var f="function"==typeof require&&requir ...
\ No newline at end of file
+/* jQuery Tree Multiselect v2.2.1 | (c) Patrick Tsai | MIT Licensed */
```
It looks like somehow you got a new version of jQuery... this should not be handled like this. There are bower and grunt projects on geonode static for that

afabiani commented on requirements.txt:
```diff
@@ -43,8 +43,13 @@ geonode-announcements==1.0.8
 geonode-dialogos==0.7
-geonode-notification==1.1.1
-geonode-user-accounts==1.0.13
+#geonode-notification==1.1.1
```
Can you cleanup the old dependencies?

afabiani commented on requirements.txt:
```diff
@@ -43,8 +43,13 @@ geonode-announcements==1.0.8
+#django-user-account
+# we can't use django-user-account until upstream merge changes for geonode.
+# this is temporary solution
+https://github.com/cezio/django-user-accounts/archive/252_255_mixed.zip#egg=django-user-accounts-2.0.2
```
Can you double check this?

afabiani commented on setup.py:
```diff
@@ -111,9 +110,14 @@
         "geonode-announcements>=1.0.8",
         "geonode-agon-ratings>=0.3.5",  # (0.3.1 in ppa) FIXME
-        "geonode-user-accounts>=1.0.13",  # (1.0.11 in ppa) FIXME
+        #"geonode-user-accounts>=1.0.13",  # (1.0.11 in ppa) FIXME
```
Can you completely remove old commented dependencies from file?

afabiani commented on setup.py:
```diff
@@ -138,4 +142,7 @@
         ],
       zip_safe=False,
+      dependency_links=[
+        'https://github.com/cezio/django-user-accounts/archive/252_255_mixed.zip#egg=django-user-accounts-2.0.2dev'
```
Is this correct?

cezio replied:
Yes, although it would be used only when running ``python setup.py install`` or ``pip install --process-dependency-links path/to/local/geonode/repo``.

afabiani commented:
@cezio thanks for your conribute. It looks a great improvement.

I did a first round of review. I also noticed that there aren't new test cases. This is why this is just a replacement for old stuff? Don'e we introduced any new functionality yet, is it correct?
